### ISSUE TITLE
RPC doesn't support "OP_RETURN <data hex bytes>" feature of GUI

### ISSUE THREAD

**Issue by pbnelson** [spesmilo/electrum#4547]
Running version 3.2.2, cloned off master as of yesterday.

Using the GUI it's possible to create OP_RETURN data transactions by simply specifying the "Pay to" as "OP_RETURN 3210", for example, storing bytes 32 10 as data in the op_return transaction. So far so good.

The issue is that this appears to work only in the GUI, not on the command line interface (CLI), or the RPC server, or the python console. In those cases, the argument parser throws `Invalid Bitcoin address or alias` when the destination is supplied in that format. For example, the commands:

    # see ~/electrum/config for rpcuser, rpcpassword and rpcport
    electrum start daemon
    electrum load_wallet  # enter walletpassword
    curl --data-binary '{"id":"curltext","method":"payto","params":{"destination":"OP_RETURN 3210", "amount":"0.000001", "fee":"0.000001", "password": "walletpassword"}}' http://rpcuser:rpcpassword==@127.0.0.1:rpcport 

Returns an error:

`Exception: ('Invalid Bitcoin address or alias', 'OP_RETURN 3210')
{"id": "curltext", "result": null, "error": {"message": "Server error: File \"/home/pbnelson/repos/electrum/lib/contacts.py\", line 95, in resolve | Exception: ('Invalid Bitcoin address or alias', 'OP_RETURN 3210')\n", "code": -32603}}`

Changing to lowercase, eliminating the hex data and just using 'op_return' alone, doesn't change the error message at all. Nor does it help to add `"nocheck":true` to the parameter list.

Similarly, using the command line interface directly also gives an error:

    electrum payto --fee 0.000001 "OP_RETURN 3210" 0.0

Returns

`  File "/home/pbnelson/.local/lib/python3.5/site-packages/electrum/contacts.py", line 95, in resolve
    raise Exception("Invalid Bitcoin address or alias", k)
Exception: ('Invalid Bitcoin address or alias', 'OP_RETURN 3210')`

Final example, this statement in the python console:

    payto('OP_RETURN 3210',[{"amount":"0.0", "fee":"0.000001"}])

Returns: 

`  File "/home/pbnelson/repos/electrum/lib/contacts.py", line 95, in resolve
    raise Exception("Invalid Bitcoin address or alias", k)
Exception: ('Invalid Bitcoin address or alias', 'OP_RETURN 3210')`

**Comment by SomberNight** [comment]
Indeed, the logic that allows sending to arbitrary scripts is currently only present for the Qt GUI.

https://github.com/spesmilo/electrum/blob/8bb59fcc3c235d971df9669afc1d7ac5a226e6ba/electrum/gui/qt/paytoedit.py#L83-L103

**Comment by pbnelson** [comment]
Fair enough, I don't want to imply that this is a "bug report", in that sense. I guess this is more of a "feature request": for the CLI/RPC interface to support the same functionality as the UI. Thanks for all your hard work on this platform, guys. It's really impressive work!
